build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/toolchain/ArchivedOracleJdkToolchainResolver.java

```diff
@@ -25,7 +25,7 @@
 
 public abstract class ArchivedOracleJdkToolchainResolver extends AbstractCustomJavaToolchainResolver {
 
-    private static final Map<Integer, String> ARCHIVED_BASE_VERSIONS = Maps.of(19, "19.0.2", 18, "18.0.2.1", 17, "17.0.7");
+    private static final Map<Integer, String> ARCHIVED_BASE_VERSIONS = Maps.of(20, "20.0.2", 19, "19.0.2", 18, "18.0.2.1", 17, "17.0.7");
```

Comment:
Do we have to manually keep this mapping? Is there no way to look this up somewhere?

Reply:
I guess one way to do this is to parse our jvm catalog. It's a good idea to automate that as especially updating jvms will be simpler. But it should be maybe done in a way where we can kind of cache the jvm catalog response instead of doing hitting this service every build. I'll will do that in a dedicated PR.